### DISCUSSION TITLE
How to avoid `mdx-code-block` showing up in TOC?

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [facebook/docusaurus#10409] [Q&A] [answered]
Hi all,

working on https://github.com/MarcoIeni/release-plz/pull/1607 I have an [example.md](https://github.com/MarcoIeni/release-plz/pull/1607/files#diff-831662cf636a0984c5ffb706cad3d958e413c192d9739a29293fbbc957524ec7) including some `mdx-code-block`  to render markdown examples and these show up in the TOC.

How can I avoid that and only have the actual `examples.md` headers in the TOC?

<img width="1255" alt="Screenshot 2024-08-16 at 2 24 32 PM" src="https://github.com/user-attachments/assets/051957ff-88cc-4299-a8fd-3dff1aa626ae">

**Comment by Josh-Cena** [comment]
`mdx-code-block` probably doesn't do what you think it does. Its job is to be completely transparent as if it wasn't there and the MDX it contains was literally written as MDX. This works around translation softwares that can't parse MDX syntax properly by pretending the MDX is just code. If you want to disable headings in TOC, one semi-hack is to replace them with JSX elements such as `<h2>Heading</h2>`.

**Reply by andreacfromtheapp** [reply]
you are most certainly right, I am possibly not using mdx as it is meant to but I needed to process markdown in a code block and it worked. if you checked the `examples.md` you can see there are two code blocks: one is "raw" markdown and the other one "rendered" to show the result of such config. I tried using `md` instead but it doesn't render it.

if there is a better way to do that I am open to it 100%

**Reply by Josh-Cena** [reply, marked as answer]
What you want is:

- Render Markdown
- Render it to _pure markup_ as if it's not actual content (i.e. no TOC, no other structural extraction we may do)

This is not a use case Docusaurus supports out of the box. What I suggested was:

Instead of this:

~~~md
```mdx-code-block
# Changelog
All notable changes to this project will be documented in this file.
The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
## [Unreleased]
```
~~~

Do this:

~~~md
<h1>Changelog</h1>
All notable changes to this project will be documented in this file.
The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
<h2>[Unreleased]</h2>
~~~

Note that this may still be a bad idea since it would break your document outline (i.e. your HTML would not be accessible). You may want to put it into an iframe or use styling to mimic headings instead.

The ability to remove certain headings from the TOC is tracked in https://github.com/facebook/docusaurus/issues/6201. However, one thing is certain: `mdx-code-block` has nothing to do with your use case and if you are not using translation softwares you can safely remove it without any other effects.

**Reply by andreacfromtheapp** [reply]
thanks @Josh-Cena , I appreciate your time and support. I will try and see how it goes. :)
